pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -89,7 +89,13 @@ func (p *Provisioner) Remove(cluster *v3.Cluster) (runtime.Object, error) {
 			break
 		}
 		if i == 3 {
-			return cluster, fmt.Errorf("failed to remove the cluster [%s]: %v", cluster.Name, err)
+			if apierrors.IsNotFound(err) {
+				logrus.Warnf("%v", err)
+				break //log the error but still remove from UI
```

Comment:
does this break do anything? you are inside of an i == 3 block, so youre on your last iteration anyway